bundles/AdminBundle/Controller/Admin/Document/PageController.php

```diff
@@ -222,6 +223,18 @@ public function saveAction(Request $request, StaticPageGenerator $staticPageGene
                 $data['staticLastGenerated'] = $staticPageGenerator->getLastModified($page);
             }
 
+            if ($page->getPrettyUrl() !== $oldPage->getPrettyUrl()
+                && empty($oldPage->getPrettyUrl()) === false
+                && empty($page->getPrettyUrl()) === false
+            ) {
+                $redirect = new Redirect();
+
+                $redirect->setSource($oldPage->getPrettyUrl());
```

Comment:
You should set the redirect type also while creating the Redirects. 
$redirect->setType(Redirect::TYPE_AUTO_CREATE);